example/postgres_arrays/postgres_arrays.pb.gorm.go

```diff
@@ -0,0 +1,469 @@
+// Code generated by protoc-gen-gogo. DO NOT EDIT.
+// source: postgres_arrays.proto
+
+/*
+Package postgres_arrays is a generated protocol buffer package.
+
+It is generated from these files:
+	postgres_arrays.proto
+
+It has these top-level messages:
+	Example
+*/
+package postgres_arrays
+
+import context "context"
+import errors "errors"
+
+import field_mask1 "google.golang.org/genproto/protobuf/field_mask"
+import gorm1 "github.com/jinzhu/gorm"
+import gorm2 "github.com/infobloxopen/atlas-app-toolkit/gorm"
+import pq1 "github.com/lib/pq"
+
+import fmt "fmt"
+import math "math"
+
+// Reference imports to suppress errors if they are not otherwise used.
+var _ = fmt.Errorf
+var _ = math.Inf
+
+type ExampleORM struct {
+	ArrayOfBools   pq1.BoolArray    `gorm:"type:bool[]"`
+	ArrayOfFloat64 pq1.Float64Array `gorm:"type:float[]"`
+	ArrayOfInt64   pq1.Int64Array   `gorm:"type:integer[]"`
+	ArrayOfString  pq1.StringArray  `gorm:"type:text[]"`
+	Description    string
+	Id             string `gorm:"type:uuid;primary_key"`
+}
+
+// TableName overrides the default tablename generated by GORM
+func (ExampleORM) TableName() string {
+	return "examples"
+}
+
+// ToORM runs the BeforeToORM hook if present, converts the fields of this
+// object to ORM format, runs the AfterToORM hook, then returns the ORM object
+func (m *Example) ToORM(ctx context.Context) (ExampleORM, error) {
+	to := ExampleORM{}
+	var err error
+	if prehook, ok := interface{}(m).(ExampleWithBeforeToORM); ok {
+		if err = prehook.BeforeToORM(ctx, &to); err != nil {
+			return to, err
+		}
+	}
+	to.Id = m.Id
+	to.Description = m.Description
+	for _, v := range m.ArrayOfBools {
+		to.ArrayOfBools = append(to.ArrayOfBools, v)
```

Comment:
(for all types of array)
Since we know in advance the size of the slice, I think it would be more efficient to `make` the slice of the length `n` as opposed to just appending `n` times which may result in multiple re-allocations. I'm not positive how the typing would work, but that might also allow for using the builtin `copy`?

Reply:
Hi @Calebjh, I've made the changes you requested. The typing works both ways, and the generated code has slightly less lines now too.